### PR TITLE
- update docs for MesonToolchain and Meson (build helper)

### PR DESCRIPTION
**Pull request by SSE4**
closes: #1975 

docs for:
https://github.com/conan-io/conan/pull/8147
https://github.com/conan-io/conan/pull/8111

### CODE REVIEW

File: creating_packages/toolchains/meson.rst [hunk 1]
```diff
@@ -29,7 +29,8 @@ The ``MesonToolchain`` can be used in the ``generate()`` method:
 
 The ``MesonToolchain`` will generate the following file during ``conan install``
 command (or before calling the ``build()`` method when the package is being
-built in the cache): ``conan_meson_native.ini``
+built in the cache): ``conan_meson_native.ini``, if doing a native build, or
+``conan_meson_cross.ini``, if doing a cross build (:ref:`cross_building_reference`).
 
 ``conan_meson_native.ini`` will contain the definitions of all the Meson properties
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:35]: Shouldn't this be?:
```suggestion
``conan_meson_native.ini`` or ``conan_meson_cross.ini`` will contain the definitions of all the Meson properties
```

**Review by czoido — APPROVED** [review]

File: creating_packages/toolchains/meson.rst [hunk 2]
```diff
@@ -70,3 +108,75 @@ With the ``MesonToolchain`` it is possible to do:
...
+Installs development files (headers, libraries, etc.). Equivalent to run :command:`meson install -C .` in the build folder.
+
+test()
+++++++
+
+.. code:: python
+
+    def test(self):
+
+Runs project's tests. Equivalent to running :command:`meson test -v -C .` in the build folder..
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:182]:
```suggestion
Runs project's tests. Equivalent to running :command:`meson test -v -C .` in the build folder.
```

File: creating_packages/toolchains/meson.rst [hunk 3]
```diff
@@ -29,7 +29,8 @@ The ``MesonToolchain`` can be used in the ``generate()`` method:
 
 The ``MesonToolchain`` will generate the following file during ``conan install``
 command (or before calling the ``build()`` method when the package is being
-built in the cache): ``conan_meson_native.ini``
+built in the cache): ``conan_meson_native.ini``, if doing a native build, or
+``conan_meson_cross.ini``, if doing a cross build (:ref:`cross_building_reference`).
```
Comment (danimtb) [inline creating_packages/toolchains/meson.rst:33]:
```suggestion
built in the cache): *conan_meson_native.ini*, if doing a native build, or
*conan_meson_cross.ini*, if doing a cross-build (:ref:`cross_building_reference`).
```

File: creating_packages/toolchains/meson.rst [hunk 4]
```diff
@@ -45,6 +46,43 @@ etc. This includes but is not limited to the following:
 
 * The Visual Studio runtime (``b_vscrt``), obtained from Conan input settings
 
+``conan_meson_cross.ini`` contains the same information as ``conan_meson_native.ini```,
+but additionally it includes information to desribe host, target and build machines (such
+as processor architecture).
```
Comment (danimtb) [inline creating_packages/toolchains/meson.rst:51]:
```suggestion
*conan_meson_cross.ini* contains the same information as *conan_meson_native.ini*,
but with additional information to describe host, target, and build machines (such
as the processor architecture).
```

File: creating_packages/toolchains/meson.rst [hunk 5]
```diff
@@ -70,3 +108,75 @@ With the ``MesonToolchain`` it is possible to do:
...
+
+configure()
++++++++++++
+
+.. code:: python
+
+    def configure(self, source_folder=None):
+
+Calls ``meson``, with the given generator and passing either ``--native-file conan_meson_native.ini``
+ (native builds) or ``--cross-file conan_meson_cross.ini`` (cross builds).
```
Comment (danimtb) [inline creating_packages/toolchains/meson.rst:153]:
```suggestion
Calls :command:`meson`, with the given generator and passing either :command:`--native-file conan_meson_native.ini`
(native builds) or :command:`--cross-file conan_meson_cross.ini` (cross builds).
```

**Review by memsharded — CHANGES_REQUESTED** [review]
All new toolchains and helpers are documented in reference -> conanfile -> tools, as ``conan.tools.meson`` in this case.

**Comment by SSE4** [comment]
> All new toolchains and helpers are documented in reference -> conanfile -> tools, as `conan.tools.meson` in this case.

as I can see `MakeToolchain`, `MSBuildToolchain`, `CMakeToolchain` are documented in `creating packages` -> `toolchain`

**Comment by memsharded** [comment]
> as I can see MakeToolchain, MSBuildToolchain, CMakeToolchain are documented in creating packages -> toolchain

@SSE4 

Those were there before creating the new ``conan.tools.xxx`` spaces. They are gradually being moved from there to the reference. Please move the Meson one there as well.

File: reference/conanfile/tools/meson.rst [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+        def generate(self):
+            tc = MesonToolchain(self)
+            tc.generate()
+
+
+The ``MesonToolchain`` will generate the following file during ``conan install``
+command (or before calling the ``build()`` method when the package is being
+built in the cache): ``conan_meson_native.ini``, if doing a native build, or
+``conan_meson_cross.ini``, if doing a cross build (:ref:`cross_building_reference`).
```
Comment (SSE4) [inline reference/conanfile/tools/meson.rst:33]:
```suggestion
built in the cache): *conan_meson_native.ini*, if doing a native build, or
*conan_meson_cross.ini*, if doing a cross-build (:ref:`cross_building_reference`).
```

File: reference/conanfile/tools/meson.rst [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+* Detection of ``buildtype`` from Conan settings
+
+* Definition of the C++ standard as necessary
+
+* The Visual Studio runtime (``b_vscrt``), obtained from Conan input settings
+
+``conan_meson_cross.ini`` contains the same information as ``conan_meson_native.ini```,
+but additionally it includes information to desribe host, target and build machines (such
+as processor architecture).
```
Comment (SSE4) [inline reference/conanfile/tools/meson.rst:51]:
```suggestion
*conan_meson_cross.ini* contains the same information as *conan_meson_native.ini*,
but with additional information to describe host, target, and build machines (such
as the processor architecture).
```

File: reference/conanfile/tools/meson.rst [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+configure()
++++++++++++
+
+.. code:: python
+
+    def configure(self, source_folder=None):
+
+Calls ``meson``, with the given generator and passing either ``--native-file conan_meson_native.ini``
+ (native builds) or ``--cross-file conan_meson_cross.ini`` (cross builds).
```
Comment (SSE4) [inline reference/conanfile/tools/meson.rst:153]:
```suggestion
Calls :command:`meson`, with the given generator and passing either :command:`--native-file conan_meson_native.ini`
(native builds) or :command:`--cross-file conan_meson_cross.ini` (cross builds).
```

File: reference/conanfile/tools/meson.rst [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+The ``Meson()`` build helper that works with the ``MesonToolchain`` is also experimental,
+and subject to breaking change in the future. It will evolve to adapt and complement the
+toolchain functionality.
+
+The helper is intended to be used in the ``build()`` method, to call Meson commands automatically
+when a package is being built directly by Conan (create, install)
+
+.. code:: python
+
+    from conans import Meson
```
Comment (memsharded) [inline reference/conanfile/tools/meson.rst:124]: Wrong import: ``from conan.tools.meson import Meson``

**Review by memsharded — APPROVED** [review]
Please fix the import and can be merged

File: reference/conanfile/tools/meson.rst (hunk 9 above)
Comment (SSE4) [inline reference/conanfile/tools/meson.rst:124]:
```suggestion
    from conan.tools.meson import Meson
```

**Pull request merged**